unpack_test.go

```diff
@@ -1563,6 +1579,97 @@ func packRar(t *testing.T, _ []archiveContent) []byte {
 	return b
 }
 
+// packRar2 creates always the same a rar archive with following files:
+// -rw-r--r--  1 503  20    27B  6 Dez 14:07 test
+// drwxr-xr-x  3 503  20    96B  6 Dez 14:08 sub/
+// -rw-r--r--  1 503  20    27B  6 Dez 14:08 sub/test
+var contentsRar2 = []archiveContent{
+	{Name: "test", Content: []byte("hello world"), Mode: 0644, AccessTime: time.Date(2024, 12, 6, 14, 7, 0, 0, time.Local), ModTime: time.Date(2024, 12, 6, 14, 8, 0, 0, time.Local), Uid: 503, Gid: 20},
+	{Name: "sub", Mode: fs.ModeDir | 0755, AccessTime: time.Date(2024, 12, 6, 14, 8, 0, 0, time.Local), ModTime: time.Date(2024, 12, 6, 14, 7, 8, 0, time.Local), Uid: 503, Gid: 20},
+	{Name: "sub/test", Content: []byte("hello world"), Mode: 0644, AccessTime: time.Date(2024, 12, 6, 14, 8, 0, 0, time.Local), ModTime: time.Date(2024, 12, 6, 14, 8, 0, 0, time.Local), Uid: 503, Gid: 20},
+}
+
+func packRar2(t *testing.T, _ []archiveContent) []byte {
+	t.Helper()
+	b, err := hex.DecodeString("526172211a0701003392b5e50a010506000501018080003afe2e322202030b9b00049b00a48302032d6c9680000104746573740a03132ff752678a911e136861736869207361797320686920746f2074686520776f726c640a7db74f802602030b9b00049b00a48302032d6c96800001087375622f746573740a031334f752672333f02b6861736869207361797320686920746f2074686520776f726c640a5311ba9e1b02030b000100ed8301800001037375620a031334f752673549ed2b1d77565103050400")
+	if err != nil {
+		t.Fatalf("error decoding rar data: %v", err)
+	}
+	return b
+}
+
+var (
+	testDataUid, testDataGid          = 1337, 42
+	testDataRootUid, testDataWheelGid = 0, 0
+	baseTime                          = time.Date(2021, 1, 1, 0, 0, 0, 0, time.Local)
+)
+
+var testCases = []struct {
+	name                  string
+	contents              []archiveContent
+	packer                func(*testing.T, []archiveContent) []byte
+	doesNotSupportModTime bool
+	doesNotSupportOwner   bool
+	expectError           bool
+	uid                   int
+	gid                   int
+}{
```

Comment:
Given that the `contents[]` entries contain UID/GID info, what is the meaning of them at the level of the whole test case? Will it always be the case that all files in an archive will have the same UID/GID pair? My instinct is to either specify this separately for each file, or just specify it for all files at the level of the test case, either of these will make the tests easier to understand I think. If there is a reason to have these at test case level as well then I think this should be documented to avoid future confusion, thanks.

Reply:
Addressed in https://github.com/hashicorp/go-extract/commit/6c8d47a3b219495b7b8e137027246829e677d335